presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/UnwrapDateFunctionInPredicate.java

```diff
@@ -0,0 +1,249 @@
+{
+    private static final String DATE_FUNCTION = "date";
+
+    private final FunctionAndTypeManager functionAndTypeManager;
+    private final StandardFunctionResolution functionResolution;
+
+    public UnwrapDateFunctionInPredicate(FunctionAndTypeManager functionAndTypeManager)
+    {
+        this.functionAndTypeManager = requireNonNull(functionAndTypeManager, "functionAndTypeManager is null");
+        this.functionResolution = new FunctionResolution(functionAndTypeManager.getFunctionAndTypeResolver());
+    }
+
+    @Override
+    public Pattern<FilterNode> getPattern()
+    {
+        return typeOf(FilterNode.class);
+    }
+
+    @Override
+    public Result apply(FilterNode node, Captures captures, Context context)
+    {
+        RowExpression predicate = node.getPredicate();
+        System.out.println("Original Predicate: " + predicate);
+
+        Optional<RowExpression> rewritten = rewritePredicate(predicate);
+
+        if (rewritten.isPresent()) {
+            System.out.println("Rewritten Predicate: " + rewritten.get());
+        } else {
+            System.out.println("No transformation applied.");
+        }
+
+        return rewritten.map(rowExpression -> Result.ofPlanNode(
+                new FilterNode(
+                        node.getSourceLocation(),
+                        node.getId(),
+                        node.getSource(),
+                        rowExpression))).orElseGet(Result::empty);
+    }
+
+    private Optional<RowExpression> rewritePredicate(RowExpression expression)
+    {
+        if (!(expression instanceof CallExpression)) {
+            return Optional.empty();
+        }
+
+        CallExpression call = (CallExpression) expression;
+        // Check if this is an equality comparison
+        if (!functionResolution.isEqualsFunction(call.getFunctionHandle())) {
+            return Optional.empty();
+        }
+
+        RowExpression left = call.getArguments().get(0);
+        RowExpression right = call.getArguments().get(1);
+
+        // Try both: function(column) = literal and literal = function(column)
+        Optional<RowExpression> rewritten = tryRewriteFunctionEqualsLiteral(left, right);
+        if (rewritten.isPresent()) {
+            return rewritten;
+        }
+
+        return tryRewriteFunctionEqualsLiteral(right, left);
+    }
+
+    private Optional<RowExpression> tryRewriteFunctionEqualsLiteral(RowExpression functionSide, RowExpression literalSide)
+    {
+        System.out.println("functionSide: " + functionSide + " (" + functionSide.getClass().getName() + ")");
+        System.out.println("literalSide: " + literalSide + " (" + literalSide.getClass().getName() + ")");
+
+        // Unwrap CAST if present on the literal side
+        if (literalSide instanceof CallExpression) {
+            CallExpression castExpression = (CallExpression) literalSide;
+            if (castExpression.getDisplayName().equalsIgnoreCase("CAST") && castExpression.getArguments().size() == 1) {
```

Comment:
Consider `OperatorType.CAST.getFunctionName()`
```suggestion
            if (castExpression.getDisplayName().equalsIgnoreCase(OperatorType.CAST.getFunctionName()) && castExpression.getArguments().size() == 1) {
```